Assets/MRTK/Providers/LeapMotion/Editor/ConfigurationChecker/LeapMotionConfigurationChecker.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.﻿
+
+using Microsoft.MixedReality.Toolkit.Utilities.Editor;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.LeapMotion
+{
+    /// <summary>
+    /// Class that checks if the Leap Motion Core assets are present and configures the project if they are.
+    /// </summary>
+    [InitializeOnLoad]
+    static class LeapMotionConfigurationChecker
+    {
+        private const string FileName = "README_BEFORE_UPDATING.txt";
+        private static readonly string[] definitions = { "LEAPMOTIONCORE_PRESENT" };
+        private static bool isLeapInProject = false;
+
+        static LeapMotionConfigurationChecker()
+        {
+            // Check if leap core is in the project
+            isLeapInProject = ReconcileLeapMotionDefine();
+
+            if (isLeapInProject)
+            {
+                RemoveTestingFolders();
```

Comment:
Do you know if this stuff is getting run on each recompile (i.e. trying to make sure we don't add a ton of duplicated work on each script recompile)

Reply:
Yeah, currently all the configuration checkers run on every recompile

Reply:
Yes I am aware the configuration checks run on every compile which is why there are guards at the beginning of each method, but I will add a more efficient solution.